spec/bson/binary_uuid_spec.rb

```diff
@@ -129,12 +129,8 @@ def make_binary(uuid_hex_str, type)
         expect(binary.to_uuid(:csharp_legacy).gsub('-', '').upcase).to eq("00112233445566778899AABBCCDDEEFF")
       end
 
-      it 'decodes as java legacy' do
-        expect(binary.to_uuid(:java_legacy).gsub('-', '').upcase).not_to eq("00112233445566778899AABBCCDDEEFF")
-      end
-
-      it 'decodes as python legacy' do
-        expect(binary.to_uuid(:python_legacy).gsub('-', '').upcase).not_to eq("00112233445566778899AABBCCDDEEFF")
+      it 'expects four dashes when output as String' do
+        expect(binary.to_uuid(:csharp_legacy).scan(/(?=-)/).count).to eq(4)
```

Comment:
Can you please change these new tests to perform equality comparisons to the expected values, rather than counting equal signs? And the removed tests should remain I expect since they don't seem to be related to the changes proposed in this PR.

Reply:
I overlooked the previous tests were doing a `not_to eq` and should not have removed them.